src/benchmarks/micro/libraries/System.Net.Sockets/Perf.Socket.cs

```diff
@@ -116,6 +118,67 @@ public async Task ReceiveAsyncThenSendAsync_SocketAsyncEventArgs()
             }
         }
 
+        [Benchmark]
+        public async Task ReceiveFromAsyncThenSendToAsync_Task()
+        {
+            ReadOnlyMemory<byte> clientBuffer = new byte[1];
+            Memory<byte> serverBuffer = new byte[1];
+
+            EndPoint ep = new IPEndPoint(IPAddress.None, 0);
+
+            var ct = new CancellationTokenSource(MaximumWait * InnerIterationCount);
+
+            for (int i = 0; i < InnerIterationCount; i++)
+            {
+                ValueTask<SocketReceiveFromResult> r = _udpServer.ReceiveFromAsync(serverBuffer, SocketFlags.None, ep, ct.Token);
+                await _udpClient1.SendToAsync(clientBuffer, SocketFlags.None, _udpServer.LocalEndPoint);
+                await r;
+                r = _udpServer.ReceiveFromAsync(serverBuffer, SocketFlags.None, ep, ct.Token);
+                await _udpClient2.SendToAsync(clientBuffer, SocketFlags.None, _udpServer.LocalEndPoint);
+                await r;
+            }
+        }
+
+        [Benchmark]
+        public async Task ReceiveFromAsyncThenSendToAsync_SocketAddress()
+        {
+            ReadOnlyMemory<byte> clientBuffer = new byte[1];
+            Memory<byte> serverBuffer = new byte[1];
+
+            SocketAddress serverSa = _udpServer.LocalEndPoint.Serialize();
+            SocketAddress receivedSa = new SocketAddress(_udpServer.AddressFamily);
+
+            var ct = new CancellationTokenSource(MaximumWait * InnerIterationCount);
+
+            for (int i = 0; i < InnerIterationCount; i++)
+            {
+                ValueTask<int> r = _udpServer.ReceiveFromAsync(serverBuffer, SocketFlags.None, receivedSa, ct.Token);
+                await _udpClient1.SendToAsync(clientBuffer, SocketFlags.None, serverSa);
+                await r;
+                r = _udpServer.ReceiveFromAsync(serverBuffer, SocketFlags.None, receivedSa, ct.Token);
+                await _udpClient2.SendToAsync(clientBuffer, SocketFlags.None, serverSa);
+                await r;
+            }
+        }
+
+        [Benchmark]
+        public void SendToThenReceiveFrom()
```

Comment:
You added async APIs for both EndPoint and SocketAddress, but sync for only EndPoint?